### ISSUE TITLE
Headers of generated .scp files broken

### ISSUE THREAD

**Issue by markusC64** [davidgiven/fluxengine#593]
New sample file:

[bsa-360.zip](https://github.com/davidgiven/fluxengine/files/9581637/bsa-360.zip)

fluxengine write ibm360 -i  bsa-360k.img -d foo.scp --drive.rotational_period_ms 200

or

fluxengine write commodore1581 -i td1581.d81 -d foo.scp --drive.rotational_period_ms 200

(the sample file from #413 )has the following issue:

* Header of generated .scp defective: end track (byte $7) and sides (byte $A) are wrong.

end track is $00 in the defective file. And sides should be "both".

This is the .scp spec: https://www.cbmstuff.com/downloads/scp/scp_image_specs.txt

I cite:

»BYTES 0x06 and 0x07 are the start track and end track bytes.  Tracks are numbered 0-167,
which is a maximum of 168 tracks (84 tracks with top/bottom).
[...]
BYTE 0x0A is the head number(s) contained in the image.  This value is either 0, 1 or 2.
If the value is 0 then both heads are contained in the image, which has always been the
default for all SCP images (except C64).  A value of 1 means just side 0 (bottom) is
contained in the image, and a value of 2 means just side 1 (top) is contained in the image.
NOTE: any program creating SCP image files should set this byte correctly.«

So in my example, byte $A should be $00, not $01. And byte $07 should be twice the number of the end track (+1 if side 1 is contained in the image), that is (if I haven't miscalulated) $9F for both the IBM360 image and for the 1581 image.

**Comment by lazypete365** [comment]
I think I can add a little bit of additional information. Both `fluxengine write` and `fluxengine read --copy-flux-to` appear to have this issue, but I have found that `fluxengine rawread` writes the header correctly.

For the same 720kb IBM floppy:
```fluxengine rawread -s drive:0 -d test_rawread.scp --drive.revolutions=5```
produces an scp file with the correct header that can be opened in other utilities like HxCFloppy Emulator:
```
53 43 50 18 FF 04 00 9F 03 00 00 00 1E B9 F6 4D
                     ^^       ^^
```
Using `fluxengine rawread -s test_rawread.flux -d test_rawread_fromflux.scp --drive.revolutions=5` with a flux file from a previous read also generates a correct scp file.

Meanwhile,
```fluxengine read ibm720 -s drive:0 -o test_copyflux.img --copy-flux-to test_copyflux.scp --drive.revolutions=5```
generates a file with the same bad values in the header as the previous post:
```
53 43 50 18 FF 04 00 00 03 00 01 00 5D A0 4D 4C
                     ^^       ^^
```

Incidentally, I also had to force `--drive.revolutions` to be 2 or more, or else the utilities I use to check scp files get very confused. Probably due to issue #518.
